src/backend/distributed/commands/citus_add_local_table_to_metadata.c

```diff
@@ -330,6 +349,68 @@ CreateCitusLocalTable(Oid relationId, bool cascadeViaForeignKeys)
 										  attnumList);
 
 	FinalizeCitusLocalTableCreation(shellRelationId, dependentSequenceList);
+
+	/*
+	 * If it's a partitioned table, we need to create Citus Local Tables from its
+	 * partitions too.
+	 */
+	CreateChildLocalTablesIfRelationIsPartitioned(shellRelationId, shardRelationId,
```

Comment:
To avoid any major changes to `CreateCitusLocalTable`, can we, for example, in `CascadeOperationForConnectedRelations` drop partitioning hierarch as we drop foreign keys, and re-create afterwards?

That should prevent any expectations around the ordering of the tables we process?